server/src/main/java/org/elasticsearch/action/admin/indices/forcemerge/ForceMergeRequest.java

```diff
@@ -158,7 +167,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeBoolean(onlyExpungeDeletes);
         out.writeBoolean(flush);
         if (out.getVersion().onOrAfter(FORCE_MERGE_UUID_VERSION)) {
-            out.writeString(forceMergeUUID);
+            out.writeString(forceMergeUUID == null ? FORCE_MERGE_UUID_NA_VALUE : forceMergeUUID);
```

Comment:
I would just use `readOptionalString` and `writeOptionalString`, no need for the extra FORCE_MERGE_UUID_NA_VALUE. It simplifies the code and no need for the extra value. Sending an extra byte on the wire does not hurt either.